deeppavlov/skills/dsl_skill/dsl_skill.py

```diff
@@ -0,0 +1,222 @@
+
+        handlers = [attribute for attribute in namespace.values() if isinstance(attribute, Handler)]
+
+        for handler in handlers:
+            if handler.state is None:
+                cls.universal_handlers.append(handler)
+            else:
+                cls.state_to_handler[handler.state].append(handler)
+
+        cls.handle = partial(DSLMeta.__handle, cls)
+        cls.__call__ = partial(DSLMeta.__handle_batch, cls)
+        cls.__init__ = partial(DSLMeta.__init__class, cls)
+        register()(cls)
+        DSLMeta.__add_to_collection(cls)
+
+    def __init__class(cls,
+                      on_invalid_command: str = "Простите, я вас не понял",
+                      null_confidence: float = 0,
+                      *args, **kwargs) -> None:
+        """
+        Initialize Skill class
+
+        Args:
+            on_invalid_command:  message to be sent on message with no associated handler
+        """
+        # message to be sent on message with no associated handler
+        cls.on_invalid_command = on_invalid_command
+        cls.null_confidence = null_confidence
+
+    def __handle_batch(cls: 'DSLMeta',
+                       utterances_batch: List[str],
+                       user_ids_batch: List[UserId]) -> Tuple[List, ...]:
+        """Returns skill inference result.
+        Returns batches of skill inference results, estimated confidence
+        levels and up to date states corresponding to incoming utterance
+        batch.
+
+        Args:
+            utterances_batch: A batch of utterances of str type.
+            user_ids_batch: A batch of user ids.
+
+        Returns:
+            response_batch: A batch of arbitrary typed skill inference results.
+            confidence_batch: A batch of float typed confidence levels for each of
+                skill inference result.
+
+        """
+        return (*map(list, zip(*starmap(cls.handle, zip_longest(utterances_batch, user_ids_batch)))),)
+
+    @staticmethod
+    def __add_to_collection(cls: 'DSLMeta') -> None:
+        """
+        Adds Skill class to Skill classes collection
+
+        Args:
+            cls: Skill class
+
+        """
+        DSLMeta.skill_collection[cls.name] = cls
+
+    @staticmethod
+    def __handle(cls: 'DSLMeta',
+                 utterance: str,
+                 user_id: UserId) -> SkillResponse:
+        """
+        Handles what is going to be after a message from user arrived.
+        Simple usage:
+        skill([<message>], [<user_id>])
+
+        Args:
+            cls: instance of callee's class
+            utterance: a message to be handled
+            user_id: id of a user
+
+        Returns:
+            result: handler function's result if succeeded
+
+        """
+        context = cls.user_to_context[user_id]
+
+        context.user_id = user_id
+        context.message = utterance
+
+        current_handler = cls.__select_handler(context)
+        current_handler.expand_context(context)
```

Comment:
This code seems to be broken since `cls.__select_handler(context)` returns a function.